src/main.c

```diff
@@ -437,8 +444,61 @@ main(int argc, char** argv)
    memset(&known_fds, 0, sizeof(known_fds));
    memset(message, 0, MISC_LENGTH);
 
+   if (directory_path != NULL)
+   {
+      if (!strcmp(directory_path, PGAGROAL_DEFAULT_CONFIGURATION_PATH))
+      {
+         pgagroal_log_warn("Using the default configuration directory %s, -D can be omitted.", directory_path);
+      }
+      if (access(directory_path, F_OK) != 0)
+      {
+         snprintf(message, MISC_LENGTH, "Configuration directory not found");
+#ifdef HAVE_LINUX
+         sd_notifyf(0, "STATUS=%s: %s", message, directory_path);
+#endif
+         errx(1, "%s: %s", message, directory_path);
+      }
+      if (access(directory_path, R_OK | X_OK) != 0)
+      {
+         snprintf(message, MISC_LENGTH, "Insufficient permissions for directory");
+#ifdef HAVE_LINUX
+         sd_notifyf(0, "STATUS=%s: %s", message, directory_path);
+#endif
+         errx(1, "%s: %s", message, directory_path);
+      }
+
+      char** config_paths[] = {
+         &configuration_path, &hba_path, &limit_path,
+         &users_path, &frontend_users_path, &admins_path, &superuser_path
+      };
+
+      char* default_files[] = {
+         PGAGROAL_DEFAULT_CONF_FILE, PGAGROAL_DEFAULT_HBA_FILE, PGAGROAL_DEFAULT_LIMIT_FILE,
+         PGAGROAL_DEFAULT_USERS_FILE, PGAGROAL_DEFAULT_FRONTEND_USERS_FILE,
+         PGAGROAL_DEFAULT_ADMINS_FILE, PGAGROAL_DEFAULT_SUPERUSER_FILE
+      };
+
+      const char* config_names[] = {
+         "configuration_path", "hba_path", "limit_path",
+         "users_path", "frontend_users_path", "admins_path", "superuser_path"
+      };
+
+      for (int i = 0; i < 7; i++)
+      {
+         if (pgagroal_normalize_path(config_paths[i], directory_path, default_files[i]) != 0)
+         {
+            snprintf(message, MISC_LENGTH, "Failed to normalize path");
+#ifdef HAVE_LINUX
+            sd_notifyf(0, "STATUS=%s: %s", message, config_names[i]);
+#endif
+            errx(1, "%s: %s", message, config_names[i]);
```

Comment:
Same

Reply:
We need also to specify what type of path is not normalized, or the user will be confused.